### PR TITLE
[12.0][IMP] cooperator_portal: backport changes from migration to 14.0

### PR DESCRIPTION
**Pull request by huguesdk**
## changes

*   improve user-facing strings.
*   improve `fr` and `fr_be` translations.
*   some formatting cleanup.

### CODE REVIEW

File: cooperator_portal/controllers/portal.py [hunk 1]
```diff
@@ -75,23 +75,21 @@ def _prepare_portal_layout_values(self):
...
+        error, error_message = super().details_form_validate(data)
+        sub_req_model = request.env["subscription.request"]
         iban = data.get("iban")
-        valid = sub_req_obj.check_iban(iban)
+        valid = sub_req_model.check_iban(iban)
 
         if not valid:
             error["iban"] = "error"
-            error_message.append(_("You iban account number is not valid"))
+            error_message.append(_("The IBAN account number is not valid."))
```
Comment (victor-champonnois) [inline cooperator_portal/controllers/portal.py:85]: Prefer "Provided IBAN is not valid." for consistency with cooperator_website's controller ?

**Review by victor-champonnois — APPROVED** [review]
LGTM

File: cooperator_portal/controllers/portal.py (hunk 1 above)
Comment (huguesdk) [reply cooperator_portal/controllers/portal.py:85]: hum… good point! why is it defined in 2 different places? i’m going to check that.

**Comment by huguesdk** [comment]
@cathLemb or @polchampion: could you please validate this functionally? it is deployed on the test servers (on all dbs, and should work on a new db also, of course). all changes are listed in the description.

**Pull request merged**